homeassistant/components/traccar/device_tracker.py

```diff
@@ -119,24 +118,24 @@ def __init__(self, api, hass, async_see, scan_interval,
         async_track_time_interval(self._hass,
                                   self._async_update,
                                   self._scan_interval)
+        if self._api.connected and not self._api.authenticated:
+            _LOGGER.error("Authentication for Traccar failed")
+            return False
+        else:
+            return True
 
     async def _async_update(self, now=None):
         """Update info from Traccar."""
-        if self.connected is None:
-            # We should only get here on the first run.
-            await self._api.test_connection()
-            self.connected = self._api.connected
-            if not self._api.connected:
-                _LOGGER.error("Not connected to Traccar.")
-
         if not self.connected:
-            _LOGGER.debug('Testing connection to Traccar.')
+            _LOGGER.debug('Testing connection to Traccar')
             await self._api.test_connection()
             self.connected = self._api.connected
             if self.connected:
-                _LOGGER.info("Connection to Traccar restored.")
-        else:
-            _LOGGER.debug('Updating device data.')
+                _LOGGER.info("Connection to Traccar restored")
+            else:
+                _LOGGER.error("Not connected to Traccar")
+        if self.connected:
```

Comment:
Instead we can return in the else case just above. Then we don't need this check and can outdent here.